sprint_3

Ilya-andreevi4 opened the pull request.

Ilya-andreevi4 committed:
create new branch and change linting

Ilya-andreevi4 committed:
cleaned up the styles

Ilya-andreevi4 committed:
Merge branch 'sprint_2' of https://github.com/Ilya-andreevi4/middle.messenger.praktikum.yandex into sprint_2

Ilya-andreevi4 committed:
Merge pull request #6 from Ilya-andreevi4/sprint_2

new sprint

Ilya-andreevi4 committed:
authApi, authController, store, router

Ilya-andreevi4 committed:
Auth for profie page

Ilya-andreevi4 committed:
fix routing

Ilya-andreevi4 committed:
fix loader

Ilya-andreevi4 committed:
add render to selected chat and fix profile page

Ilya-andreevi4 committed:
hotfix

Ilya-andreevi4 committed:
Add close a modals with click on bg

Ilya-andreevi4 committed:
Chat Api with create/read/delete

Ilya-andreevi4 committed:
Chats controller

Ilya-andreevi4 committed:
add messenger and web socket

Ilya-andreevi4 committed:
Some fix

Ilya-andreevi4 committed:
-logs

Ilya-andreevi4 committed:
ReadMe

Ilya-andreevi4 committed:
modified:   README.md

gennady-bars commented on src/components/chat-actions-popup/popup-form-chat-action.hbs:
```diff
@@ -0,0 +1,8 @@
+    {{{deleteLink}}}
+  {{/each}}
+
+</form>
```
В проекте в некоторых файлах есть ошибка `EOF`. Для `git` важно наличие **пустой строки** в конце файла. Подробнее тут: https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log Можно в настройки IDE добавить `insert_final_newline = true`

gennady-bars commented on src/utils/fetch.ts:
```diff
@@ -5,138 +5,134 @@ enum METHODS {
+// }
+
+export default class HTTPTransport {
+  static API_URL = "https://ya-praktikum.tech/api/v2";
```
* Отлично, что базовый урл вынесен отдельно, чтобы не дублировать его

gennady-bars commented on src/utils/fetch.ts:
```diff
@@ -5,138 +5,134 @@ enum METHODS {
+    return this.request(this.endpoint + currentPath, { method: METHODS.GET });
+  }
+
+  public post<Response = void>(path: string, data?: JsonObject | FormData): Promise<Response> {
```
**Можно лучше**
Если типизация повторяется в каждом методе, то нужно типизировать сам метод, а не дублировать типизацию аргументов. Код станет чище

```jsx

// создаем тип метода
type HTTPMethod<Response = void> = (url: string, options?: JsonObject | FormData) => Promise<Response >

export class HTTPTransport {

  // используем тип и удаляем дублирование в аргументах
  get: HTTPMethod = (url, options = {}) => (
    this.request(url, {...options, method: METHODS.GET}, options.timeout)
  )
  // используем тип и удаляем дублирование в аргументах
  put: HTTPMethod = (url, options = {}) => (
    this.request(url, {...options, method: METHODS.PUT}, options.timeout)
  )
  // используем тип и удаляем дублирование в аргументах
  post: HTTPMethod = (url, options = {}) => (
    this.request(url, {...options, method: METHODS.POST}, options.timeout)
  )
  // используем тип и удаляем дублирование в аргументах
  delete: HTTPMethod = (url, options = {}) => (
    this.request(url, {...options, method: METHODS.DELETE}, options.timeout)
  )

```

Ilya-andreevi4 replied:
Пофиксил критические ошибки. С типизацией методов и с избавлением от any планирую заняться завтра. Благодарю за ревью!

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная




* `Readme` хорошо оформлен
* Проект задеплоен и отлично работает
* Хорошая структура папок и файлов
* Отлично, что тип пропсов получаете из дженерика в классе `Block`
* Отлично, что указываете понятный текст ошибки под инпутом
* Отлично, что базовый урл вынесен отдельно, чтобы не дублировать его

но есть некоторые недочеты: 


* Скрин https://disk.yandex.ru/i/gQLTzO10ZH-mjw Я могу добавить **пустое** сообщение в чат, а валидация не должна мне этого давать
* По заданию **эндпоинты** в проекте должны быть определённые. Скрин из задания https://disk.yandex.ru/i/5EcJYattAj8NJA 
* По заданию требуется: При обновлении страницы с определённым URL должна отображаться та же самая страница. Если  я **перезагружаю** страницу в профиле, то должен попадать в профиль.
* Когда добавляю или удаляю пользователя, он не появляется (не исчезает). Приходится перезагружать страницу. И аватар тоже не сразу отображается в профиле после изменения
* Должен быть скрипт `dev` для запуска в режиме разработки без сборки. А то не могу запустить проект
* В проекте в некоторых файлах есть ошибка `EOF`. Для `git` важно наличие **пустой строки** в конце файла. Подробнее тут: https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log Можно в настройки IDE добавить `insert_final_newline = true`


**Можно лучше**

* Если типизация повторяется в каждом методе, то нужно типизировать сам метод, а не дублировать типизацию аргументов. Код станет чище
* Попробуйте избавиться от типов `any` в коде. Хотя бы часть из них можно заменить на `unknown`


Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте **работоспособность проекта** и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

Удачного рефакторинга кода.

Ilya-andreevi4 committed:
modified:   .eslintignore
	modified:   .eslintrc.json
	modified:   .prettierrc
	modified:   .stylelintignore
	modified:   package-lock.json
	modified:   package.json
	modified:   src/api/AuthAPI.ts
	modified:   src/api/ChatsAPI.ts
	modified:   src/api/ProfileAPI.ts
	modified:   src/components/add-files-popup/index.ts
	modified:   src/components/avatar/index.ts
	modified:   src/components/button/index.ts
	modified:   src/components/chat-actions-popup/index.ts
	modified:   src/components/chat-info/index.ts
	modified:   src/components/field/field.hbs
	modified:   src/components/field/field.pcss
	modified:   src/components/field/index.ts
	modified:   src/components/icon/index.ts
	modified:   src/components/input/index.ts
	modified:   src/components/link/index.ts
	modified:   src/components/message-input-form/index.ts
	modified:   src/components/message-input-form/message-input-form.pcss
	modified:   src/components/message/index.ts
	modified:   src/components/nav/index.ts
	modified:   src/components/popup-item/index.ts
	modified:   src/controllers/AuthContoller.ts
	modified:   src/controllers/ChatController.ts
	modified:   src/controllers/MessagesController.ts
	modified:   src/controllers/ProfileController.ts
	modified:   src/index.ts
	modified:   src/layouts/form/index.ts
	modified:   src/layouts/loader/index.ts
	modified:   src/pages/login-page/index.ts
	modified:   src/pages/main-page/chat-container/chat-header/index.ts
	modified:   src/pages/main-page/chat-container/chat-main/index.ts
	modified:   src/pages/main-page/chat-container/index.ts
	modified:   src/pages/main-page/index.ts
	modified:   src/pages/main-page/side-menu/friends-container/index.ts
	modified:   src/pages/main-page/side-menu/index.ts
	modified:   src/pages/main-page/side-menu/profile-block/index.ts
	modified:   src/pages/not-found-page/index.ts
	modified:   src/pages/profile-page/index.ts
	modified:   src/pages/registration-page/index.ts
	modified:   src/pages/server-error-page/index.ts
	modified:   src/utils/Block.ts
	modified:   src/utils/Interfaces.ts
	modified:   src/utils/Router.ts
	modified:   src/utils/Store.ts
	modified:   src/utils/WSTransport.ts
	modified:   src/utils/fetch.ts
	modified:   src/utils/helpers.ts
	modified:   src/utils/media-exports.ts
	modified:   src/utils/page-fields.ts
	modified:   src/utils/routes.ts

Ilya-andreevi4 committed:
hotfix

Ilya-andreevi4 committed:
Fix eof

Ilya-andreevi4 committed:
hotfix

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.



* По заданию требуется: При обновлении страницы с определённым URL должна отображаться та же самая страница. Если  я **перезагружаю** страницу в профиле, то должен попадать в профиль. На `Нетлифай` это не работает сейчас

Ilya-andreevi4 committed:
fix redirects on netlify
